Sources/ArcGISToolkit/Components/FormView/FormInputs/InputFooter.swift

```diff
@@ -17,23 +17,204 @@ import SwiftUI
+        )
+    }
+    
+    /// Determines whether an error is showing in the footer.
+    var isShowingError: Bool {
+        element.isEditable && primaryError != nil && model.previouslyFocusedFields.contains(element)
+    }
+    
+    /// The alloable range of number of characters in the input.
+    var lengthRange: ClosedRange<Int>? {
+        if let input = element.input as? TextAreaFormInput {
+            return input.minLength...input.maxLength
+        } else if let input = element.input as? TextBoxFormInput, element.fieldType == .text {
+            return input.minLength...input.maxLength
+        } else {
+            return nil
+        }
+    }
+    
+    /// Text indicating a field's value must be within the allowed length range.
+    var lengthRangeMessage: Text {
+        Text(
+            "Enter \(lengthRange!.lowerBound) to \(lengthRange!.upperBound) characters",
+            bundle: .toolkitModule,
+            comment: """
+                         Text indicating a field's value must be within the 
+                         allowed length range. The first and second parameter
+                         hold the minimum and maximum length respectively.
+                         """
+        )
+    }
+    
+    /// Text indicating a field's maximum number of allowed characters.
+    var maximumLengthMessage: Text {
+        Text(
+            "Maximum \(lengthRange!.upperBound) characters",
+            bundle: .toolkitModule,
+            comment: "Text indicating a field's maximum number of allowed characters."
+        )
+    }
+    
+    /// Text indicating a field's maximum number of allowed characters.
+    var minimumLengthMessage: Text {
+        Text(
+            "Minimum \(lengthRange!.lowerBound) characters",
+            bundle: .toolkitModule,
+            comment: "Text indicating a field's minimum number of allowed characters."
+        )
+    }
+    
+    /// The error to display for the input. If the element is not focused and has a required error this will be
+    /// the primary error. Otherwise the primary error is the first error in the element's set of errors.
+    var primaryError: ArcGIS.FeatureFormError? {
+        let elementErrors = model.validationErrors[element.fieldName] as? [ArcGIS.FeatureFormError]
+        if let requiredError = elementErrors?.first(where: { $0 == .featureFormFieldIsRequiredError }), model.focusedElement != element {
+            return requiredError
+        } else {
+            return elementErrors?.first(where: { $0 != .featureFormFieldIsRequiredError })
+        }
+    }
+}
+
+private extension RangeDomain {
+    /// String representations of the minimum and maximum value of the range domain.
+    var displayableMinAndMax: (min: String, max: String)? {
+        if let min = minValue as? Double, let max = maxValue as? Double {
+            return (min.formatted(.number.precision(.fractionLength(1...3))), max.formatted(.number.precision(.fractionLength(1...3))))
```

Comment:
Does this range for precision seem okay?

Reply:
> What are you trying to accomplish?

The goal is to formate Double based RangeDomains with trailing zeros even if there was no fractional portion (e.g. minimum of 3 shows as 3.0).

We have a UI test that [expects this](https://devtopia.esri.com/runtime/common-toolkit/blob/main/designs/Forms/FormsTestDesign.md#test-case-14-numeric-text-input:~:text=the%20text%20field-,Expectation%3A%20the%20helper%20text%20now%20says%20%22Enter%20value%20from%202.0%20to%205.0%22,-clear%20the%20current).

I think I've found the right solution which is to use [PartialRangeFrom](https://developer.apple.com/documentation/swift/partialrangefrom) starting at one `1...` which from my experiment seems like it will append anything from a `.0` up to the level of accuracy stored in the Double.

```swift
min.formatted(.number.precision(.fractionLength(1...))
```